snapshot/bundlestore/store.go

```diff
@@ -14,9 +29,10 @@ type StoreRead interface {
 }
 
 // Write operations on store, limited to a one-shot writing operation since bundles are immutable.
+// If ttl config is nil then the store will use its defaults.
 type StoreWrite interface {
 	// Does a streaming write of the given bundle. There is no concept of partial writes (partial=failed).
-	Write(name string, data io.Reader) error
+	Write(name string, data io.Reader, ttl *TTLValue) error
```

Comment:
TTLValue uses time.Time so it self-documents as expiration time which I think is sufficient.